src/rank_llm/demo/rerank_rank_gpt.py

```diff
@@ -0,0 +1,33 @@
+import sys
+import os
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+parent = os.path.dirname(SCRIPT_DIR)
+parent = os.path.dirname(parent)
+sys.path.append(parent)
+
+from rank_llm.retrieve.retriever import Retriever
+from rank_llm.rerank.rankllm import PromptMode
+from rank_llm.rerank.rank_gpt import SafeOpenai
+from rank_llm.rerank.reranker import Reranker
+from rank_llm.retrieve_and_rerank import get_api_key
+
+# By default uses BM25 for retrieval
+dataset_name = "dl19"
+retrieved_results = Retriever.from_dataset_with_prebuit_index(dataset_name)
+agent = SafeOpenai("gpt-3.5-turbo", 4096, PromptMode.RANK_GPT, 0, keys=get_api_key())
```

Comment:
Had a question about this; the issue says "no need to populate parameters with default values", however it seems rankGPT agent requires these parameters.

Reply:
Please add the promptmode and fewshot values as default params in the SafeOpenai class. model name and context_size are fine to be passed here, since openai offers different prompt sizes for different models.